### PR TITLE
Enable eslint and auto formatting

### PR DESCRIPTION
**Pull request by justinchuby**
I enabled eslint and auto formatting in VS Code and took the liberty to pick a style guide that is closest to the existing style.
Replaces #57

### CODE REVIEW

**Comment by tintinweb** [comment]
Hey @justinchuby, thanks for the PR 🙌

would be great if you could exclude the files that were changed in the eslint run from the actual config (i.e. only include eslintrc, .vscode/*, package.json). that would make it easier for us to run it locally once the current feature branches/PRs are settled and merge the PR.

thanks,
tin

**Comment by justinchuby** [comment]
Done!

**Comment by tintinweb** [comment]
@bigbug you know more about typescript than I do :) can you give this a look? I remember the changelog of this was pretty significant and we might get to review it with most of the recent features now merged, thx! 🙏

**Comment by bigbug** [comment]
@tintinweb: Done. All ts files have been linted and the errors have been fixed.

**Comment by justinchuby** [comment]
Thank you! I updated from master and resolved conflicts in `package-lock.json`

**Comment by bigbug** [comment]
I still have to check whether and how the style rules apply to the javascript part in the Webview.
@justinchuby  Thank you for resolving the conflict!

**Comment by bigbug** [comment]
content-folder has been linted initially. However, the structure of our current content folder and how we include the js files does appear to make linting not possible and a lot of disables would be necessary. Therefore linting has been disable for the content-folder until a fundamental redesign (probably to typescript) has taken place.
From my point of view: PR approved!

**Review by bigbug — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -138,16 +138,19 @@
...
-        "eslint": "^7.32.0",
+        "eslint": "^8.8.0",
+        "eslint-config-airbnb-base": "^15.0.0",
+        "eslint-plugin-import": "^2.25.4",
         "glob": "^7.1.7",
         "mocha": "^9.1.1",
         "raw-loader": "^4.0.2",
         "ts-loader": "^9.2.5",
         "typescript": "^4.4.3",
+        "vscode": "^0.9.9",
```
Comment (tintinweb) [inline package.json:153]: is this really needed? seems to be deprecated? https://www.npmjs.com/package/vscode

Comment (justinchuby) [reply package.json:153]: Removed this dependency and updated the lock file

Comment (bigbug) [reply package.json]: I think it got into the code base when I experimented with upcoming type-errors. The proper usage was to include "@types/vscode". I think the removal should be fine. Good catch!

**Review by bigbug — APPROVED** [review]

**Comment by tintinweb** [comment]
excellent, thanks for all the work on this 🤗

**Pull request merged**